test/e2e_node/gpus.go

```diff
@@ -151,6 +151,29 @@ var _ = framework.KubeDescribe("GPU [Serial]", func() {
 
 			By("Checking if the pod outputted Success to its logs")
 			framework.ExpectNoError(f.PodClient().MatchContainerOutput(podFailure.Name, podFailure.Name, "Success"))
+
+			By("checking allocatable gpus")
+			gpusAvailableBefore := getGPUsAvailable(f)
+
+			By("restarting the kubelet")
+			if stdout, err := exec.Command("sudo", "systemctl", "restart", "kubelet").CombinedOutput(); err != nil {
+				framework.Logf("Failed to trigger kubelet restart with systemd/systemctl: %v, stdout: %q", err, string(stdout))
+				if stdout, err = exec.Command("sudo", "service", "kubelet", "restart").CombinedOutput(); err != nil {
+					framework.Failf("Failed to trigger kubelet restart with upstart/service: %v, stdout: %q", err, string(stdout))
+				}
+			}
+			time.Sleep(20 * time.Second)
+
+			By("expecting the same number of allocatable gpus")
+			gpusAvailableAfter := getGPUsAvailable(f)
```

Comment:
I don't think this PR achieves your intent which IIUC is that of testing if kubelet keeps track of allocated GPUs across restarts.
Why not, schedule a pod that consumes all GPUs, then restart kubelet, and then try scheduling another pod that wants a GPU and validate that the new pod stays pending?